File: frontend/app/models/place.js
import DS from 'ember-data';

export default DS.Model.extend({
  name: DS.attr('string'),
  name2: DS.attr('string'),
  place_type: DS.attr('string'),
  street: DS.attr('string'),
  post_code: DS.attr('number'),
  address: DS.attr('string'),
  cost: DS.attr('number'),
  currency: DS.attr('string'),
  hours: DS.attr('string'),
  website: DS.attr('string'),
  mediaDesc: DS.attr('string'),
  mediaSrc: DS.attr('string'),
  notes: DS.attr('string'),
  latitude: DS.attr(),
  longitude: DS.attr(),
  city_id: DS.belongsTo('city'),
  country_id: DS.belongsTo('country')
});

File: app/models/place.rb
class Place < ActiveRecord::Base
  before_validation :set_address
  belongs_to :country
  belongs_to :city

  geocoded_by :address
  after_validation :geocode

  def set_address
    if self.village.nil?
      self.address = self.street.to_s + ", " + self.post_code.to_s + " " + self.city.name.to_s + ", " + self.country.name.to_s
    else
      self.address = self.street.to_s + ", " + self.post_code.to_s + " " + self.village.to_s + ", " + self.country.name.to_s
    end
  end
end
